clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -423,11 +433,12 @@ private boolean maybeSendTransactionalRequest() {
             if (!accumulator.flushInProgress())
                 accumulator.beginFlush();
         }
+    }
 
-        TransactionManager.TxnRequestHandler nextRequestHandler = transactionManager.nextRequestHandler(accumulator.hasIncomplete());
-        if (nextRequestHandler == null)
-            return false;
-
+    /**
+     * Enqueue a FindCoordinatorRequest if needed, otherwise send the next request
+     */
+    private boolean maybeSendTransactionalRequest(TransactionManager.TxnRequestHandler nextRequestHandler) {
         AbstractRequest.Builder<?> requestBuilder = nextRequestHandler.requestBuilder();
         while (!forceClose) {
```

Comment:
Not necessarily something we have to do here, but I think we should be able to get rid of this loop and just rely on the next iteration of `runOnce()` to handle retries.